rosbag2_transport/src/rosbag2_transport/topic_filter.cpp

```diff
@@ -129,36 +149,78 @@ bool TopicFilter::take_topic(
     return false;
   }
 
-  if (!record_options_.topics.empty() && !topic_in_list(topic_name, record_options_.topics)) {
-    return false;
-  }
+  if (!is_service_event_topic) {
+    if (!record_options_.all_topics &&
+      record_options_.topics.empty() &&
+      record_options_.regex.empty() &&
+      !record_options_.include_hidden_topics)
+    {
+      return false;
+    }
 
-  std::regex exclude_regex(record_options_.exclude);
-  if (!record_options_.exclude.empty() && std::regex_search(topic_name, exclude_regex)) {
-    return false;
-  }
+    if (!record_options_.topics.empty() && !topic_in_list(topic_name, record_options_.topics)) {
+      return false;
+    }
 
-  std::regex include_regex(record_options_.regex);
-  if (
-    !record_options_.all &&  // All takes precedence over regex
-    !record_options_.regex.empty() &&  // empty regex matches nothing, but should be ignored
-    !std::regex_search(topic_name, include_regex))
-  {
-    return false;
-  }
+    std::regex exclude_topics_regex(record_options_.exclude_topics);
+    if (!record_options_.exclude_topics.empty() &&
+      std::regex_search(topic_name, exclude_topics_regex))
+    {
+      return false;
+    }
 
-  if (!has_single_type(topic_name, topic_types)) {
-    return false;
+    std::regex include_regex(record_options_.regex);
+    if (!record_options_.all_topics &&  // All takes precedence over regex
+      !record_options_.regex.empty() &&  // empty regex matches nothing, but should be ignored
+      !std::regex_search(topic_name, include_regex))
+    {
+      return false;
+    }
+  } else {
+    if (!record_options_.all_services &&
+      record_options_.services.empty() &&
+      record_options_.regex.empty())
+    {
+      return false;
+    }
+
+    if (!record_options_.services.empty() &&
+      !service_in_list(topic_name, record_options_.services))
+    {
+      return false;
+    }
+
+    // Convert service event topic name to service name
+    auto service_name =
+      rosbag2_cpp::service_event_topic_name_to_service_name(topic_name);
```

Comment:
Comments from MichaelOrlov

```suggestion
    auto service_name =  rosbag2_cpp::service_event_topic_name_to_service_name(topic_name);
```
Nitpick. It could fit in one line.

Reply:
Updated